tests/mcmc/test_sampling.py

```diff
@@ -449,6 +452,7 @@ def normal_logprob(self, x):
     def test_univariate_normal(
         self, algorithm, initial_position, parameters, num_sampling_steps, burnin
     ):
+        algorithm = eval(algorithm)
```

Comment:
Not sure I understand this change, seems not needed?

Reply:
Because `blackjax.hmc` (or any other kernel constructor) is now a function, it creates an object with a specific location in memory when evaluated. If this object is created outside the test function `test_univariate_normal` then calls to the chex variant function `inference_loop` have a different call sign for each of the devices/cores (because each call sign has a different memory location), throwing an error when collecting tests. The error looks like `ERROR collecting gw1. Different tests were collected between gw0 and gw1`.